File: 0841_KeysandRooms/KeysandRooms.js
var canVisitAllRooms = function(rooms) {
    const visited = new Set([0])
    const queue = [0]
    while (queue.length){
        let des = queue.pop()
        for (let key of rooms[des]) {
            if (visited.has(key)) continue
            visited.add(key)
            queue.push(key)
        }
    }
    return visited.size === rooms.length
};

File: 0841_KeysandRooms/KeysandRooms.py
class Solution:
    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        explored = set([0])
        queue = collections.deque([0])
        while queue:
            room = queue.popleft()
            for key in rooms[room]:
                if key in explored: continue
                queue.append(key)
                explored.add(key)
        return True if len(explored) == len(rooms) else False